javalib/src/main/scala/java/util/concurrent/ConcurrentSkipListSet.scala

```diff
@@ -0,0 +1,404 @@
+package java.util.concurrent
+
+import java.util._
+import java.lang.Comparable
+import scala.collection.mutable
+import scala.math.Ordering
+
+import scala.collection.JavaConversions._
+
+class ConcurrentSkipListSet[E >: Null] protected (ordering: Ordering[E], provided: Boolean) 
+    extends AbstractSet[E]
+    with NavigableSet[E]
+    with Cloneable
+    with Serializable { self =>
+
+  def this() =
+    this(ConcurrentSkipListSet.defaultOrdering[E], false)
+
+  def this(comparator: Comparator[E])  = 
+    this(Ordering.comparatorToOrdering(comparator), true)
+
+  def this(sortedSet: SortedSet[E]) = {
+  	this(Ordering.comparatorToOrdering(sortedSet.comparator()), true)
+    addAll(sortedSet)
+  }
+
+  def this(collection: java.util.Collection[E]) = {
+  	this(ConcurrentSkipListSet.defaultOrdering[E], false)
+    addAll(collection)
+  }
+
+  def this(comparator: Comparator[E], collection: Collection[E]) = {
+    this(Ordering.comparatorToOrdering(comparator), true)
+    addAll(collection)
+  }
+
+  implicit object BoxOrdering extends Ordering[Box[E]] {
+    def compare(a: Box[E], b:Box[E]): Int = ordering.compare(a.inner, b.inner)
+  }
+
+  private val inner: mutable.SortedSet[Box[E]] = new mutable.TreeSet[Box[E]]()
+
+  override def isEmpty(): Boolean =   
+    inner.isEmpty
+
+  def iterator(): Iterator[E] = {
```

Comment:
I don't think this implementation of `iterator()` satisfies the *weekly consistent* property it's supposed to have. From [the JavaDoc of `ConcurrentSkipListSet`](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentSkipListSet.html):

> Iterators are weakly consistent, returning elements reflecting the state of the set at some point at or since the creation of the iterator. They do not throw `ConcurrentModificationException`, and may proceed concurrently with other operations.

This is hard to fix, other than creating an immediate copy of the full set into an immutable data structure every time a new iterator is created. Instead, maybe use an immutable TreeSet for `inner`, but make it a `var`? This way the `iterator()` method can capture the current value of `inner` at the time it is created, and then work with that.